app/models/chargeback_configured_system.rb

```diff
@@ -0,0 +1,97 @@
+class ChargebackConfiguredSystem < Chargeback
+  set_columns_hash(
+    :configured_system_id    => :integer,
+    :configured_system_name  => :string,
+    :configured_system_uid   => :string,
+    :configured_system_guid  => :string,
+    :provider_name           => :string,
+    :tenant_name             => :string,
+    :provider_uid            => :string,
+    :cpu_allocated_metric    => :float,
+    :cpu_allocated_cost      => :float,
+    :cpu_cost                => :float,
+    :fixed_compute_1_cost    => :float,
+    :fixed_compute_2_cost    => :float,
+    :memory_allocated_cost   => :float,
+    :memory_allocated_metric => :float,
+    :memory_cost             => :float,
+    :total_cost              => :float
+  )
+
+  def self.build_results_for_report_ChargebackConfiguredSystem(options)
+    @report_user = User.find_by(:userid => options[:userid])
+
+    @configured_systems = nil
+    build_results_for_report_chargeback(options)
+  end
+
+  def self.where_clause(records, options, region)
+    scope = records.where(:resource_type => "ConfiguredSystem")
+    if options[:tag] && (@report_user.nil? || !@report_user.self_service?)
+      scope_with_current_tags = scope.where(:resource => ConfiguredSystem.find_tagged_with(:any => @options[:tag], :ns => '*'))
+      scope.for_tag_names(options[:tag].split("/")[2..-1]).or(scope_with_current_tags)
+    else
+      scope.where(:resource => configured_systems(region))
+    end
+  end
+
+  def self.extra_resources_without_rollups(region)
+    scope = ConfiguredSystem.eager_load(:hardware, :taggings, :tags)
+
+    if @options[:tag] && (@report_user.nil? || !@report_user.self_service?)
+      scope.find_tagged_with(:any => @options[:tag], :ns => '*')
+    else
+      scope.where(:id => configured_systems(region))
+    end
+  end
+
+  def self.report_static_cols
+    %w[configured_system_name]
+  end
+
+  def self.report_col_options
+    {
+      "cpu_allocated_cost"      => {:grouping => [:total]},
+      "cpu_allocated_metric"    => {:grouping => [:total]},
+      "cpu_cost"                => {:grouping => [:total]},
+      "fixed_compute_metric"    => {:grouping => [:total]},
+      "fixed_compute_1_cost"    => {:grouping => [:total]},
+      "fixed_compute_2_cost"    => {:grouping => [:total]},
+      "fixed_cost"              => {:grouping => [:total]},
+      "memory_allocated_cost"   => {:grouping => [:total]},
+      "memory_allocated_metric" => {:grouping => [:total]},
+      "memory_cost"             => {:grouping => [:total]},
+      "total_cost"              => {:grouping => [:total]}
+    }
+  end
+
+  def self.configured_systems(region)
+    @configured_systems ||= {}
+    @configured_systems[region] ||=
+      begin
+        if @options[:entity_id]
+          ConfiguredSystem.where(:id => @options[:entity_id])
+        elsif @options[:tag]
+          configured_systems = ConfiguredSystem.find_tagged_with(:all => @options[:tag], :ns => '*')
```

Comment:
`configured_systems ` is not needed here:
```ruby
...
elsif @options[:tag]
  ConfiguredSystem.find_tagged_with(:all => @options[:tag], :ns => '*')
else
...
```

Reply:
This has been addressed in latest PR.